nDPI/osx/README.md

```diff
@@ -0,0 +1,32 @@
+# Building nDPI Static and Dynamic LIB for MacOS
+
+Note: Libraries need to be built for both Intel and Arm processors.
+
+Use repo: https://github.com/infiotinc/nDPI
+Branch Use: infiot-ndpi-3.0
+
+How to Build:
+- cd nDPI
+- ./bootstrap.sh
+- ./configure
+- make
+
+Use above steps to build libraies on Intel anf Arm based macbooks to generate libraries for
+both platforms.
+
+- Libraies can be found in below path:
+    nDPI/src/lib/libndpi.so
+    nDPI/src/lib/libndpi.a
+
+Use below commands to build universal (x85_64 + arm64) libraries:
+    lipo x86_64-osx/libndpi.a arm64-osx/libndpi.a -create -output universal/libndpi.a
+    lipo x86_64-osx/libndpi.so arm64-osx/libndpi.so -create -output universal/libndpi.so 
+
+Use below command to check the architecture type of the libraries:
+    lipo -info universal/libndpi.*
+
+
+3rd June 2023:
+- osx nDPI libraries are built from below repo and branch:
```

Comment:
@ns-vishalk the folders should be organized by `<arch>-<os>` right? should this not be `arm64-osx` and `x86_64-osx`?

Reply:
it is named based on what we follow in Netskope at other places. All macOS files is kept under 'osx' folder. But in case if you and Anand agree to keep arch name in folder then I would recommend 'universal-osx', as the libraries are joined together to support both the platforms. we don't need to maintain libraries of two platforms separately.

Reply:
> in macOS world we use the term for the combined binaries/libraries for both x86_64 and arm64 as 'universal'.

sounds good @ns-vishalk then let us just name it as `osx-universal` to make it clear if that is ok with you